### PR TITLE
Implemented OrbitCalculations and Prediciton

### PR DESCRIPTION
**Pull request by haennes**
Things added:

- OrbitCalculation (GravityObjects attract each other)
- Orbits are predicted
- Prediction gets shown and added to a Path (Godot 4.x: Path3d)

### CODE REVIEW

File: scripts/GravityObject/GravityObjects.gd [hunk 1]
```diff
@@ -0,0 +1,47 @@
+extends RigidBody
+
+##### Declare this in an Autoload later   (from)
+
+var G = 0.6
```
Comment (Pawel82S) [inline scripts/GravityObject/GravityObjects.gd]: This should be:
`
const G := 0.6
`

File: scripts/GravityObject/GravityObjects.gd [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	var force = Vector3.ZERO
+	var GravityObjects = get_tree().get_nodes_in_group("GravityEmitter")
+	for x in GravityObjects.size():
+		var GravityObject : RigidBody = GravityObjects[x]
+		if GravityObject == self:
+			continue
+		var direction = translation.direction_to(GravityObject.translation)
+		var squaredistance = translation.distance_squared_to(GravityObject.translation)
+		force = direction * G * GravityObject.mass * mass / squaredistance
+		print("x : %s dir : %s squaredistance : %s force : %s" % [x,direction,squaredistance,force])
```
Comment (Pawel82S) [inline scripts/GravityObject/GravityObjects.gd]: Missing spaces between parameters in quare brackets for better readablity:
`
[x, direction, squaredistance, force]
`

File: scripts/OrbitPredictor/OrbitPrediction.gd [hunk 3]
```diff
@@ -0,0 +1,95 @@
+extends Spatial
+
+
+var orbit_animation : Animation
+var time : float = 0
+export var seemslikeorbit : bool = false
+export var AnimationPlayerchild = "AnimationPlayer"
+var GravityObjects
```
Comment (Pawel82S) [inline scripts/OrbitPredictor/OrbitPrediction.gd]: This is array and should be initialized like array.
`
var gravity_objects := []
`

**Review by Pawel82S — CHANGES_REQUESTED** [review]
In general:
- function and variable names should be all small leters written in snake_case
- all constants in CAPITAL_LETTERS
- spaces between mathematical operators and variables constants ie: `var1 * var2` not `var1*var2`
- spaces between parameters, ie: `parameter1, parameter2` not `parameter1,parameter2`
It's all written in CoC.md file.

File: scripts/OrbitPredictor/OrbitPrediction.gd (hunk 3 above)
Comment (haennes) [reply scripts/OrbitPredictor/OrbitPrediction.gd]: Good idea but I think
var Gravity objects : Array
Would be even better

Comment (Pawel82S) [reply scripts/OrbitPredictor/OrbitPrediction.gd]: every variable has some initialization like "time" line 5, so arrays and dictionaries should be initialized in similar way for consistency sake and written in snake_case not CamelCase convention.
`
var example_array := []
var example_dictionary := {}
`

Comment (haennes) [reply scripts/OrbitPredictor/OrbitPrediction.gd]: Sorry didnt notice it is a outdated file this isn't used anymore

**Comment by haennes** [comment]
Nice

**Pull request merged**